### PR TITLE
[MC-1340] BMC Helix ITSM plugin (6 actions)

### PR DESCRIPTION
**Pull request by mgrandys-r7**
### Description

Describe the proposed changes:

  - Implement BMC Helix ITSM plugin with 6 actions

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/BMC Helix https://example.com Step name: createIncident\nCreating an incident with following parameters: \n{'First_Name': '007', 'Last_Name': 'Agent', 'Description': 'Create Incident Unit test', 'Impact': '4-Minor/Localized', 'Urgency': '2-High', 'Status': 'Assigned', 'Reported Source': 'Direct Input', 'Service_Type': 'User Service Restoration'}...\nGetting authentication token...\n",
    "meta": {},
    "output": {
      "incidentNumber": "INC000000000720"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/bmc_helix_itsm:1.0.0 run < tests/createIncident.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/BMC Helix https://example.com Step name: createProblemInvestigation\nCreating a problem investigation with following parameters: \n{'Site Group': 'United States', 'Region': 'Americas', 'Site': 'Houston Support Center', 'Description': 'Create Problem Investigation Unit Test', 'Company': 'MyCompany', 'Last Name': 'Agent', 'First Name': '007', 'Urgency': '3-Medium', 'Impact': '3-Moderate/Limited', 'Investigation Driver': 'High Impact Incident', 'Support Company Pblm Mgr': 'MyCompany', 'Support Organization Pblm Mgr': 'MyCompany Support Org', 'Support Group Name Requester': 'MyCompany Support Group1', 'Problem Manager Login': '007 Agent', 'Assigned Support Company': 'MyCompany', 'Assigned Support Organization': 'MyCompany Support Org', 'Assigned Group Pblm Mgr': 'MyCompany Support Group1', 'Assignee Pblm Mgr': '007 Agent', 'z1D_Action': 'PROBLEM'}...\nGetting authentication token...\n",
    "meta": {},
    "output": {
      "problemInvestigationNumber": "PBI000000000312"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/bmc_helix_itsm:1.0.0 run < tests/createProblemInvestigation.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/BMC Helix https://example.com Step name: createTask\nCreating an task related to INC000000000526 incident with following parameters: \n{'Summary': 'UnitTestTaskSummary', 'RootRequestMode': 'Real', 'Notes': 'Unit test task notes', 'TaskType': 'Manual', 'TaskName': 'my name task unit test', 'Priority': 'Medium', 'Location Company': 'MyCompany', 'RootRequestFormName': 'HPD:Help Desk'}...\nGetting details about INC000000000526 incident...\nGetting authentication token...\n",
    "meta": {},
    "output": {
      "taskId": "TAS000000000207"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/bmc_helix_itsm:1.0.0 run < tests/createTask.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/BMC Helix https://example.com Step name: getIncident\nGetting details about INC000000000526 incident...\nGetting authentication token...\n",
    "meta": {},
    "output": {
      "incident": {
        "assignedGroup": "MyCompany Support Group1",
        "assignedSupportCompany": "MyCompany",
        "assignedSupportOrganization": "MyCompany Support Org",
        "assignee": "007 Agent",
        "description": "my updated description plugin action",
        "entryId": "INC000000000474",
        "impact": "4-Minor/Localized",
        "incidentNumber": "INC000000000526",
        "lastModifiedBy": "007 Agent",
        "lastModifiedDate": "https://example.com",
        "lastResolvedDate": "https://example.com",
        "ownerGroup": "MyCompany Support Group1",
        "ownerSupportCompany": "MyCompany",
        "ownerSupportOrganization": "MyCompany Support Org",
        "priority": "Medium",
        "reportedDate": "https://example.com",
        "reportedSource": "Direct Input",
        "resolution": "Resolve through Rest API",
        "status": "Resolved",
        "statusReason": "Monitoring Incident",
        "submitDate": "https://example.com",
        "submitter": "007 Agent",
        "timeZone": "(GMT-06:00) Central Time (US \u0026 Canada)",
        "urgency": "2-High"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/bmc_helix_itsm:1.0.0 run < tests/getIncident.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/BMC Helix https://example.com Step name: getIncidentWorkInformation\nGetting work information about INC000000000526 incident...\nGetting details about INC000000000526 incident...\nGetting authentication token...\n",
    "meta": {},
    "output": {
      "worklogs": [
        {
          "assignWorklogFlag": "No",
          "description": "Assignment Change",
          "incidentNumber": "INC000000000526",
          "lastModifiedBy": "Remedy Application Service",
          "lastModifiedDate": "https://example.com",
          "secureWorkLog": "Yes",
          "status": "Enabled",
          "submitDate": "https://example.com",
          "submitter": "Remedy Application Service",
          "viewAccess": "Internal",
          "workLogDate": "https://example.com",
          "workLogId": "WLG000000000547",
          "workLogSubmitDate": "https://example.com",
          "workLogSubmitter": "Remedy Application Service",
          "workLogType": "Assignment Change"
        },
        {
          "assignWorklogFlag": "No",
          "description": "Status Change Reason",
          "incidentNumber": "INC000000000526",
          "lastModifiedBy": "007 Agent",
          "lastModifiedDate": "https://example.com",
          "secureWorkLog": "Yes",
          "status": "Enabled",
          "submitDate": "https://example.com",
          "submitter": "007 Agent",
          "viewAccess": "Internal",
          "workLogDate": "https://example.com",
          "workLogId": "WLG000000000549",
          "workLogSubmitDate": "https://example.com",
          "workLogSubmitter": "007 Agent",
          "workLogType": "Status Change"
        }
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/bmc_helix_itsm:1.0.0 run < tests/getIncidentWorkInformation.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/BMC Helix https://example.com Step name: modifyIncident\nModifying an INC000000000526 incident with following parameters: \n{'Description': 'my updated description plugin action', 'Status': 'Resolved', 'Status_Reason': 'Monitoring Incident', 'Resolution': 'Resolve through Rest API'}...\nGetting authentication token...\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/bmc_helix_itsm:1.0.0 run < tests/modifyIncident.json
</summary>
</details>

### UI

Screenshots of the plugin being used in Komand for validating use go below this line.

### CODE REVIEW

**Comment by mgrandys-r7** [comment]
Plugin Valiadtor is failing because of missing icon.png file

**Review by dsliwinski-r7 — APPROVED** [review]

File: plugins/bmc_helix_itsm/plugin.spec.yaml [hunk 1]
```diff
@@ -0,0 +1,638 @@
+plugin_spec_version: v2
+extension: plugin
+products: [insightconnect]
+name: bmc_helix_itsm
+title: BMC Helix ITSM
+description: BMC Helix IT Service Management is a cloud-native SaaS solution leveraging AI and automation to accelerate IT team results
+version: 1.0.0
+supported_versions: ["BMC Helix ITSM 21.30.00"]
+vendor: rapid7
+support: rapid7
```
Comment (dlaverty-r7) [inline plugins/bmc_helix_itsm/plugin.spec.yaml:10]: Should this have cloud_ready enabled ?

**Review by igorski-r7 — APPROVED** [review]

File: plugins/bmc_helix_itsm/plugin.spec.yaml (hunk 1 above)
Comment (dlaverty-r7) [reply plugins/bmc_helix_itsm/plugin.spec.yaml:10]: Following discussion with @wjohnstone-r7 , this plugin will not be cloud enabled.

**Comment by dlaverty-r7** [comment]
@mgrandys-r7 , can you take a look at the unit test failures on this one please? 
'ImportError: Start directory is not importable'

**Comment by mgrandys-r7** [comment]
> @mgrandys-r7 , can you take a look at the unit test failures on this one please? 'ImportError: Start directory is not importable'

@dlaverty-r7 Done. There is only as issue with missing icons

**Review by dlaverty-r7 — APPROVED** [review]

**Pull request merged**